### PR TITLE
[qxlsx/1.4.5] Bump version

### PR DESCRIPTION
**Pull request by MartinDelille**
- [qxlsx/1.4.5] Bump version
- Move 1.4.3 patches to subfolder
- Use export_conandata_patches
- Bump qt dependency to 5.15.7

Specify library name and version:  **qxlsx/1.4.5**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by MartinDelille** [comment]
> What about to drop 1.4.3 and related patches?

I could but I read somewhere that we generally keep the last 3 minor versions of a recipe.

File: recipes/qxlsx/all/conanfile.py [hunk 1]
```diff
@@ -25,34 +26,42 @@ class QXlsxConan(ConanFile):
...
     def requirements(self):
-        self.requires("qt/5.15.5")
+        self.requires("qt/5.15.7")
 
     def layout(self):
-        cmake_layout(self)
+        cmake_layout(self, src_folder="src")
+
+    def build_requirements(self):
+        self.tool_requires("cmake/3.25.0")
```
Comment (SpaceIm) [inline recipes/qxlsx/all/conanfile.py:51]: Why is it required unconditionally while it was building fine in c3i for older versions?

Comment (MartinDelille) [reply recipes/qxlsx/all/conanfile.py]: Sorry you already review this here: https://github.com/conan-io/conan-center-index/pull/14562#discussion_r1039379016

I just add the check here: 4f8882579

File: recipes/qxlsx/all/conanfile.py [hunk 2]
```diff
@@ -25,34 +26,44 @@ class QXlsxConan(ConanFile):
...
+        self.requires("qt/5.15.7")
 
     def layout(self):
-        cmake_layout(self)
+        cmake_layout(self, src_folder="src")
+
+    def build_requirements(self):
+        if Version(self.version) >= "1.4.4":
+            # at least CMake 3.16
+            self.tool_requires("cmake/3.25.0")
```
Comment (SpaceIm) [inline recipes/qxlsx/all/conanfile.py:53]:
```suggestion
    def _cmake_new_enough(self, required_version):
        try:
            import re
            from io import StringIO
            output = StringIO()
            self.run("cmake --version", output=output)
            m = re.search(r"cmake version (\d+\.\d+\.\d+)", output.getvalue())
            return Version(m.group(1)) >= required_version
        except:
            return False

    def build_requirements(self):
        if Version(self.version) >= "1.4.4" and not self._cmake_new_enough("3.16"):
            self.tool_requires("cmake/3.25.0")
```

File: recipes/qxlsx/all/conanfile.py [hunk 3]
```diff
@@ -25,34 +26,44 @@ class QXlsxConan(ConanFile):
         "fPIC": True
     }
 
+    @property
+    def _qt_version(self):
+        return Version(self.deps_cpp_info["qt"].version).major
```
Comment (SpaceIm) [inline recipes/qxlsx/all/conanfile.py:31]:
```suggestion
        return Version(self.dependencies["qt"]ref..version).major
```

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**